### ISSUE TITLE
App thinning

### ISSUE THREAD

**Issue by Kentzo** [Kentzo/ShortcutRecorder#74]
Hopefully sometime in the future xcassets will allow only include resources that match the deployment target condition.

**Comment by Kentzo** [comment]
As of 10.14 there is still no app thinning for macOS.

**Comment by lwouis** [comment]
Hey @Kentzo, could you clarify the scope of this ticket? Maybe I can help here. What exactly should match the deployment target condition?

**Comment by Kentzo** [comment]
First it needs to be verified that App thinning is available for macOS 11. Then resources (styles) for older OSes should be excluded.

**Comment by lwouis** [comment]
I think App Thinning is mostly used when discussing Distribution-side thinning. Specifically, you download an app from the Mac App Store, and depending on your local environment, Apple is going to [strip away necessary content from that download](https://medium.com/@ankur.s20/implementing-app-thinning-in-your-project-step-by-step-tutorial-ios-app-b3cfd139896d).

I was [looking into such things for AltTab](https://github.com/lwouis/alt-tab-macos/issues/158), and realized that it's only supported for MAS apps. This means that if you want to have more optimized builds for specific platforms, you need to handle that yourself in your distribution channel.

That being said, SR is a framework, not an app. Maybe there is support for that through SPM or Cocoapods? I think you should clarify which channel you are interested in, and which targets you would optimize for. You mentioned styles. Is there anything else?

**Comment by Kentzo** [comment]
Last time I checked App Thinning was not supported on macOS.

**Comment by Kentzo** [comment]
You're correct that thinning happens before distribution. However, the project must be built and properly annotated to support that.
